src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.util.ArrayList;
+
+/**
+ * The operations of task list.
+ */
+public class TaskList {
+    ArrayList<Task> tasks; //The tasklist.
+
+    /**
+     * Initiates the object with empty list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Initiates the object with tasklist.
+     *
+     * @param tasks The task list.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Deletes task from task list.
+     *
+     * @param n The order should be delete.
+     * @return The task be deleted.
+     * @throws DukeException When n > tasklist's size.
+     */
+    public Task deleteTask(int n) throws DukeException {
+        if (n > tasks.size()) {
+            throw new DukeException(" ☹ OOPS!!! Do not exist that task.");
+        }
+        Task removeItem = tasks.get(n - 1);
```

Comment:
Maybe you could do something like "Task removeItem = tasks.remove(n - 1)" ? Cuz remove() will return the item that is removed from the list. This can improve the performance of the method as we only need to traverse the list once in this case.